ci/cleanup-pr-previews

```diff
@@ -0,0 +1,252 @@
+    2>/dev/null || true)
+
+if [[ -z "$PR_FOLDERS" ]]; then
+    echo -e "${YELLOW}[INFO]${NC} No PR preview folders found in gh-pages branch"
+    exit 0
+fi
+
+echo -e "${GREEN}[INFO]${NC} Found $(echo "$PR_FOLDERS" | wc -l) PR preview folders"
+
+# Check each PR folder
+FOLDERS_TO_REMOVE=()
+TOTAL_FOLDERS=0
+OPEN_PRS=0
+
+while IFS= read -r folder; do
+    if [[ -z "$folder" ]]; then
+        continue
+    fi
+
+    TOTAL_FOLDERS=$((TOTAL_FOLDERS + 1))
+
+    # Extract PR number from folder name (pr-XXXXX -> XXXXX)
+    PR_NUMBER="${folder#pr-}"
+
+    echo -e "${YELLOW}[CHECK]${NC} Checking PR #${PR_NUMBER}..."
+
+    # Check PR status using GitHub API
+    PR_STATUS=$(gh api repos/"${REPOSITORY}"/pulls/"${PR_NUMBER}" \
+        --header "Accept: application/vnd.github+json" \
+        --jq '.state' 2>/dev/null || echo "not_found")
+
+    case "$PR_STATUS" in
+        "open")
+            echo -e "${GREEN}[KEEP]${NC} PR #${PR_NUMBER} is still open"
+            OPEN_PRS=$((OPEN_PRS + 1))
+            ;;
+        "closed")
+            echo -e "${RED}[REMOVE]${NC} PR #${PR_NUMBER} is closed"
+            FOLDERS_TO_REMOVE+=("$folder")
+            ;;
+        "not_found")
+            echo -e "${RED}[REMOVE]${NC} PR #${PR_NUMBER} not found (may have been deleted)"
+            FOLDERS_TO_REMOVE+=("$folder")
+            ;;
+        *)
+            echo -e "${YELLOW}[UNKNOWN]${NC} PR #${PR_NUMBER} has unexpected status: ${PR_STATUS}"
+            ;;
+    esac
+done <<< "$PR_FOLDERS"
+
+# Summary
+echo ""
+echo -e "${YELLOW}[SUMMARY]${NC}"
+echo "Total PR preview folders: ${TOTAL_FOLDERS}"
+echo "Open PRs: ${OPEN_PRS}"
+echo "Folders to remove: ${#FOLDERS_TO_REMOVE[@]}"
+
+if [[ ${#FOLDERS_TO_REMOVE[@]} -eq 0 ]]; then
+    echo -e "${GREEN}[INFO]${NC} No cleanup needed - all preview folders correspond to open PRs"
+    exit 0
+fi
+
+# List folders to be removed
+echo ""
+echo -e "${YELLOW}[FOLDERS TO REMOVE]${NC}"
+for folder in "${FOLDERS_TO_REMOVE[@]}"; do
+    pr_num="${folder#pr-}"
+    echo "  - $folder (PR #${pr_num})"
+done
+
+# Perform cleanup or show what would be done
+echo ""
+if [[ "$DRY_RUN" == "true" ]]; then
+    echo -e "${YELLOW}[DRY RUN]${NC} Would remove ${#FOLDERS_TO_REMOVE[@]} folders (run without --dry-run to actually remove)"
+else
+    echo -e "${RED}[CLEANUP]${NC} Proceeding to remove ${#FOLDERS_TO_REMOVE[@]} folders..."
+
+    # Create a git worktree for gh-pages branch
+    TEMP_DIR="./gh-pages-cleanup"
+    rm -rf ${TEMP_DIR}
+    mkdir ${TEMP_DIR}
+
+    # Cleanup function to properly remove worktree and temp directory
+    cleanup_worktree() {
+        cd - >/dev/null 2>&1 || true  # Go back to original directory
+        if [[ -n "$TEMP_DIR" && -d "$TEMP_DIR" ]]; then
+            git worktree remove "$TEMP_DIR" --force >/dev/null 2>&1 || true
+        fi
+        rm -rf "$TEMP_DIR" >/dev/null 2>&1 || true
+    }
+    trap cleanup_worktree EXIT
```

Comment:
This cleanup should not be executed if the changes made to the local worktree is not yet pushed, otherwise all local changes are lost after the script ends.